### PR TITLE
[flutter_tools] remove build script from tool package

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

With the main flutter run workflow moved to the frontend_server, the tool no longer needs to directly depend on many build_runner APIs. The build_script itself is launched in a separate dart process, and currently only for test --platform=chrome. To reduce the dependencies of the flutter tool (moving from direct to dev_dependencies), we can create a separate package for this build script until we decide on a path forward for  https://github.com/flutter/flutter/issues/50594

### CODE REVIEW

File: packages/flutter_tools/pubspec.yaml [hunk 1]
```diff
@@ -121,11 +103,26 @@ dev_dependencies:
   build_vm_compilers: 1.0.4
   build_test: 0.10.12+1
 
+  bazel_worker: 0.1.23+1 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
```
Comment (jonahwilliams) [inline packages/flutter_tools/pubspec.yaml]: A lot of these libraries are still depended on via build_test, but that might be removable in the future

File: packages/_flutter_web_build_script/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+name: _flutter_web_build_script
```
Comment (zanderso) [inline packages/_flutter_web_build_script/pubspec.yaml]: Consider adding an explanatory README.md for this package that points at any relevant github issues.

Comment (jonahwilliams) [reply packages/_flutter_web_build_script/pubspec.yaml]: Done

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 3]
```diff
@@ -187,19 +187,19 @@ class BuildDaemonCreator {
...
     WebTestTargetManifest testTargets,
   }) {
+    // The build script is stored in an auxillary package to reduce
+    // dependencies of the main tool.
     final String flutterToolsPackages = globals.fs.path.join(
       Cache.flutterRoot,
       'packages',
-      'flutter_tools',
+      '_flutter_web_build_script',
       '.packages',
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: As discussed offline, test for the existence of this file and fail in a nice way if it's not there.

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 4]
```diff
@@ -187,19 +187,19 @@ class BuildDaemonCreator {
     bool initializePlatform,
     WebTestTargetManifest testTargets,
   }) {
+    // The build script is stored in an auxillary package to reduce
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: nit: auxiliary

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 5]
```diff
@@ -187,19 +187,19 @@ class BuildDaemonCreator {
...
+    // dependencies of the main tool.
     final String flutterToolsPackages = globals.fs.path.join(
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: This variable name is no longer accurate

**Review by yjbanov — APPROVED** [review]
LGTM if LGT @zanderso

**Review by zanderso — APPROVED** [review]
still lgtm

**Pull request merged**